gateway/handler_unixfs__redirects.go

```diff
@@ -118,44 +148,34 @@ func (i *handler) handleRedirectsFileRules(w http.ResponseWriter, r *http.Reques
 	return false, "", nil
 }
 
-func (i *handler) getRedirectRules(r *http.Request, redirectsFilePath ipath.Resolved) ([]redirects.Rule, error) {
-	// Convert the path into a file node
-	node, err := i.api.GetUnixFsNode(r.Context(), redirectsFilePath)
+// getRedirectRules fetches the _redirects file corresponding to a given path and returns the rules
+// Returns whether _redirects was found, the rules (if they exist) and if there was an error (other than a missing _redirects)
+// If there is an error returns (false, nil, err)
+func (i *handler) getRedirectRules(r *http.Request, redirectsPath ImmutablePath) (bool, []redirects.Rule, error) {
+	// Check for _redirects file.
+	// Any path resolution failures are ignored and we just assume there's no _redirects file.
+	// Note that ignoring these errors also ensures that the use of the empty CID (bafkqaaa) in tests doesn't fail.
+	_, redirectsFile, err := i.api.Get(r.Context(), redirectsPath)
 	if err != nil {
-		return nil, fmt.Errorf("could not get _redirects: %w", err)
+		// TODO: this adds a requirement on the Get interface
```

Comment:
I think this got resolved:
- we have `isErrNotFound` which takes care of all "not found" errors), and  we have 
- implementer of `Get` can return  `NewErrorResponse(err, http.StatusNotFound)`
```suggestion
```